StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift

```diff
@@ -47,7 +47,19 @@ extension PaymentSheet {
                     // Note: Shipping address is passed for Apple Pay in STPApplePayContext+PaymentSheet.swift.
                     // For other payment methods, get shipping address from configuration.
                     let shippingAddress = configuration.shippingDetails()?.stpAddress
-                    preparePaymentMethodHandler(paymentMethod, shippingAddress)
+
+                    // Try to create a radar session for the payment method before calling the handler
+                    configuration.apiClient.createSavedPaymentMethodRadarSession(paymentMethodId: paymentMethod.stripeId) { _, error in
+                        // If radar session creation fails, just continue with the payment method directly
+                        if let error = error {
+                            // Log the error but don't fail the payment
+                            let errorAnalytic = ErrorAnalytic(event: .savedPaymentMethodRadarSessionFailure, error: error)
+                            STPAnalyticsClient.sharedClient.log(analytic: errorAnalytic, apiClient: configuration.apiClient)
+                        }
+
+                        // Call the handler regardless of radar session success/failure
+                        preparePaymentMethodHandler(paymentMethod, shippingAddress)
+                    }
                     completion(.completed, STPAnalyticsClient.DeferredIntentConfirmationType.completeWithoutConfirmingIntent)
```

Comment:
seems like the completion should be called directly after preparepaymentMethodHandler?